File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer ,  Integer > mp = new HashMap<>() ;
        for(int i = 0 ; i < nums.length ; i ++)
        {
            mp.put(nums[i] , i);
        }
        
        int [] arr = new int[2];
        for(int i = 0; i < nums.length ; i ++)
        {
            if(mp.containsKey(target - nums[i]))
            {
                if(i != mp.get(target - nums[i])){
                    arr[0] = i;
                    arr[1] = mp.get(target - nums[i]) ;
                    break;
                }
            }
        }
        return arr;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& arr, int target) 
    {
        vector<pair<int ,int >> v ;
        for(int i = 0 ; i < arr.size() ; i ++)
        {
            v.push_back({arr[i] , i}) ;
        }
        
        sort(v.begin() , v.end()) ;
        
        int start = 0 , end = v.size() - 1;
        while(start <= end)
        {
            if(v[start].first + v[end].first == target)
            {
                return {v[start].second , v[end].second};
            }
            else if(v[start].first + v[end].first > target)
            {
                end --;
            }
            else {start++;}
        }
        
        return {};
        
    }
};